modeling-data-relationships/modeling-answers.md

```diff
@@ -0,0 +1,121 @@
+
+Since the question requires that we have also the grades for each student taking a class, we will need a `grades` entity as well. 
+
+**What fields/attributes will each entity need**
+
+A `students` entity will need the following attributes `id`, `first_name`, and `last_name`. 
+
+The `courses` entity will need the following attributes: `id`, and `course_name`. 
+
+The `grades` entity will need the following attributes: `id`, `course_id`, and `student_id`.
+
+**What data types do we need to use?**
+
+For the `students`, `id` should be an INTEGER and a PRIMARY KEY, `first_name` and `last_name` should be VARCHAR.
+
+For `courses`, `id` should be an INTEGER and a PRIMARY KEY, `course_name` should be a VARCHAR and `credits` as an INTEGER. 
+
+For `grades`, `id` should be an INTEGER and the PRIMARY KEY.`course_id` is a FOREIGN KEY that REFERENCES `courses.id`,and `student_id` is also a FOREIGN KEY that REFERENCES `students.id`. Both of these FOREIGN KEYS should be an INTEGER. Finally, grade is a CHAR(1), that represents the single letter grade.
+
+**what relationships exist between entities?**
+
+As mentioned in a previous queston in this section a), every row in `students` can be taking multiple courses. Every class the student takes will have a grade.
+
+Conversely, every class will have multiple students, and each student in the course will have a grade.
+
+Because of this relationship, the grades table will have a grade entry for each course for every student that is taking a course.
+
+**How should those relationships be represented in tables?**
+
+The `students` table should have a PRIMARY KEY to represent every student, in this case `id`. 
+
+Similarly, the `courses` table should have a PRIMARY KEY that represents the course itself, in this case `id` within this table. 
+
+Now, `grades`, also should have a PRIMARY KEY that represents every grade granted to a given student within a given course. We'll also call this the `id`. However, the relationship between a student and their grades is many to one, likewise, the relationship between a course and the grades is one to many. The grades table will be the intermediary table between courses and students, with the grade data found in the intermediary table. These relationships are represented via a FOREIGN KEY represented in the `grades` table.
+
+In the `grades` table, the attribute `course_id` and `student_id` would be the FOREIGN keys that link this table with `students` and `courses`. Also these foreign key would also link `students` and `courses` together via `grades`
+
+b) the UML Diagram is found at this link: https://drive.google.com/open?id=0B0Xx9V-Et7D2eTBVZGpRQlQ0akE
+
+
+3)
+
+a)
+```
+SELECT students.first_name, students.last_name 
+FROM students
+JOIN grades ON grades.student_id = students.id
+JOIN courses ON courses.id = grades.course_id
+WHERE courses.name  = 'Digital Circuits Lab';
+```
+b)
+
+```
+SELECT grades.grade, COUNT(*) AS amount
```

Comment:
you're missing a filter

Reply:
Ahh... For a particular class! Missed that. :+1: